dashboard/app/models/script_level.rb

```diff
@@ -744,6 +744,25 @@ def level_deprecated?
     level&.deprecated?
   end
 
+  # WARNING: Do NOT reuse this trashy little method. It is fragile English-only string comparison
+  # written for a very specific use case - logging analytics for the CSA '24-'25 AI Tutor pilot,
+  # in which the level progression naming conventions follow a very specific pattern aligned
+  # with the PRIMM pedagogical approach.
+  #
+  # If the concept of a "progression type" becomes more general, or you're tempted to use this method,
+  # consider a more robust solution such as creating a new property that can be designated by a
+  # Levelbuilder, similar to how we designate assessment levels.
+  def primm_progression_type
+    progression_name = properties["progression"]
+    substring = progression_name.split(":").first if progression_name
+    return "predict_and_run" if substring&.include?("Predict and Run")
+    return "investigate_and_modify" if substring&.include?("Investigate and Modify")
+    return "practice" if substring&.include?("Practice")
+    return "project" if substring&.include?("Project")
+    return "assessment" if substring&.include?("Check for Understanding")
```

Comment:
Given that AI Tutor is "off" (or should be) for all assessment levels I don't expect us to ever hit this case, but if it does show up in logging then we know something has gone sideways and we'll have the level id to backtrace the mismatch.